src/SW_SoilWater.c

```diff
@@ -560,6 +560,35 @@ void SW_WaterBalance_Checks(SW_ALL* sw, LOG_INFO* LogInfo)
 }
 #endif
 
+/**
+ * @brief Initialize all possible pointers in SW_SOILWAT to NULL
+ *
+ * @param[in,out] SW_SoilWat Struct of type SW_SOILWAT containing
+ * 		soil water related values
+*/
+void SW_SWC_init_ptrs(SW_SOILWAT* SW_SoilWat) {
+	OutPeriod pd;
+
+	// Initialize output structures:
+	ForEachOutPeriod(pd)
+	{
+		SW_SoilWat->p_accu[pd] = NULL;
+		if (pd > eSW_Day) {

```

Comment:
We can do `SW_*_init_ptrs()` but not `SW_*_deconstruct()`. This is due to what `average_for()` does with the first period in `p_oagg`. It sets the pointer in the array of index `eSW_Day` to `p_accu`, eliminating its NULL status. Then, when we deallocate, it'll deallocate `p_accu`'s memory and try to free that instead, causing a problem. We could set `p_oagg` to NULL again (of index `eSW_Day`) once the simulation is over, but this may be unnecessary when we can keep what we currently have.